Sources/AblyAssetTrackingPublisher/DefaultPublisher.swift

```diff
@@ -851,6 +856,13 @@ extension DefaultPublisher {
             self.delegate?.publisher(sender: self, didUpdateResolution: event.resolution)
         }
     }
+    
+    private func notifyTrackablesChanged(trackables: Set<Trackable>) {
+        performOnMainThread { [weak self] in
```

Comment:
Just a question, wouldn't this be a bit more suited to put on code where main thread is performing ? Such as in UI code

Reply:
Not sure I understand the question, but generally - `performOnMainThread` runs the code on the main, UI thread:
```swift
DispatchQueue.main.async(execute: operation)
```
We're running similar delegate methods on this same thread already as well, like:
notifyDelegateEnhancedLocationChanged,
notifyDelegateConnectionStateChanged,
notifyDelegateResolutionUpdate

etc.

Reply:
I meant that I would expect the delegates  to perform this on main thread but not the `DefaultPublisher` itself. Unless operations are confined on main thread. But I see that this pattern has been used elsewhere in this class so I think it's ok for now.